src/atopile/front_end.py

```diff
@@ -838,29 +724,53 @@ def _new_node(
             # Descend into building the superclass. We've got no information
             # on when the super-chain will be resolved, so we need to promise
             # that this current blockdef will be visited as part of the init
-            new_node, supers_ = self._new_node(base_class, [item] + promised_supers)
+            if super_ctx:
+                from_ctxs_ = from_ctxs + [super_ctx]
+            else:
+                from_ctxs_ = from_ctxs
+
+            result = self._new_node(
+                base_class,
+                promised_supers=[item] + promised_supers,
+                from_ctxs=from_ctxs_,
+            )
 
             # HACK: promised_supers is falsey a hack way to check
             # that we're only looking at the top-node w/ promised supers
             if not promised_supers and (block_type.COMPONENT() or block_type.MODULE()):
-                new_node.add(has_ato_cmp_attrs())
+                result[0].add(has_ato_cmp_attrs())
 
-            return new_node, supers_
+            return result
 
         # This should never happen
         raise ValueError(f"Unknown item type {item}")
 
-    def _init_node(self, stmt_ctx: ParserRuleContext, ref: Ref) -> L.Node:
+    def _init_node(
+        self,
+        node_type: ap.BlockdefContext | Type[L.Node],
+        stmt_ctx: ParserRuleContext | None,
+    ) -> L.Node:
         """Kind of analogous to __init__ in Python, except that it's a factory"""
-        new_node, promised_supers = self._new_node(
-            self._get_referenced_class(stmt_ctx, ref)
+        new_node, promised_supers, from_ctxs = self._new_node(
+            node_type,
+            promised_supers=[],
+            from_ctxs=[stmt_ctx] if stmt_ctx is not None else [],
         )
 
-        with self._node_stack.enter(new_node):
-            for super_ctx in promised_supers:
-                self.visitBlock(super_ctx.block())
+        backup_ctx_stack = self._traceback_stack.copy()
+        self._traceback_stack.extend(from_ctxs)
+        try:
+            with self._node_stack.enter(new_node):
+                for super_ctx in promised_supers:
+                    self.visitBlock(super_ctx.block())
+                    self._traceback_stack.pop()
+        finally:
+            # Guarentee that we're gonna end up
+            # with the original context stack
+            self._traceback_stack = backup_ctx_stack
 
-        new_node.add(from_dsl(stmt_ctx))
+        if stmt_ctx is not None:
```

Comment:
No `stmt_ctx` is provided in the file's context, because it was a CLI command that invoked the build/traceback before that